### ISSUE TITLE
Improved pages module configuration

### ISSUE THREAD

**Issue by Laruxo** [nuxt/framework#10062]
### Describe the feature

Currently, the pages module has a very limited configuration option. It can be enabled/disabled and we can change the dir name. I would like to improve this to bring it on par with the configuration available for components and composables.

Example: in the following directory structure I want to automatically load only Vue files (could be customizable with a glob pattern).
```
pages
  somePage
    index.vue
    pageSpecificHelper.ts
    pageAsyncData.ts
    generated
      GQLGeneratedTypes.ts
```
Currently, I have a couple of workarounds like adding `-` prefixes, or `ignore` config (though it affects all auto-imports).

I would be happy to work on this myself. The idea I have would be to allow `pages` config to be an object with an option for `pattern` (or use the existing `ScanDir` interface for more configuration) and then change [this line in resolvePagesRoutes](https://github.com/nuxt/framework/blob/main/packages/nuxt/src/pages/utils.ts#L38) with fallback to the current pattern.

---

More broadly speaking, I imagine that a more consistent and granular configuration of auto-imports would benefit many bigger projects. But this would be a bigger change affecting components, composables, plugins, and pages configuration. 

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by manniL** [comment]
You can already adapt it via `ignore` in `nuxt.config.ts` and apply a glob to e.g. ignore all non-vue-files. See https://github.com/nuxt/framework/blob/9b80b35c1695d0ea733f37fa14e2b91a0da84d4f/packages/schema/src/config/common.ts#L337-L345

**Comment by Laruxo** [comment]
@manniL thank you for your response. As I wrote above, I know about this solution, but it has a big drawback of affecting all auto-imports. E.g. I still want to use regular TS files for composables.

**Comment by manniL** [comment]
But not if you specify the pages folder via `ignore` property ☺️

**Comment by Laruxo** [comment]
True, my current solution has this line in the `ignore` array: `'pages/**/*.ts'`

Right now I don't have another use case for my proposal. But I still think the configuration would benefit from more consistent and granular options along the lines of [ScanDir interface](https://github.com/nuxt/framework/blob/main/packages/schema/src/types/components.ts#L15). What do you think?
